docs/CNDP_README.md

```diff
@@ -0,0 +1,421 @@
+
+```
+diff --git a/conf/cndp_upf_1worker.jsonc b/conf/cndp_upf_1worker.jsonc
+index 5c2fdaf..8d7b8da 100644
+--- a/conf/cndp_upf_1worker.jsonc
++++ b/conf/cndp_upf_1worker.jsonc
+@@ -84,9 +84,9 @@
+     //    description   - (O) The description, 'desc' can be used as well
+     // CNDP lports for Access network followed by lports for Core network.
+     "lports": {
+-        "enp134s0:0": {
++        "ens803f2:0": {
+             "pmd": "net_af_xdp",
+-            "qid": 22,
++            "qid": 10,
+             "umem": "umem0",
+             "region": 0,
+             "busy_poll": true,
+@@ -94,9 +94,9 @@
+             "busy_timeout": 20,
+             "description": "Access LAN 0 port"
+         },
+-        "enp136s0:0": {
++        "ens803f3:0": {
+             "pmd": "net_af_xdp",
+-            "qid": 22,
++            "qid": 10,
+             "umem": "umem0",
+             "region": 1,
+             "busy_poll": true,
+```
+
+5) Modify the script [docker_setup.sh](../scripts/docker_setup.sh) and update the access and core interface names (s1u, sgi), access/core interface mac addresses and neighbor gateway interfaces mac addresses. This should match the access/core netdev interface name used in jsonc file in previous step. In our example test setup, neighbor mac address (n-s1u, n-sgi) corresponds to access/core interfaces used by packet generator in system 2 to send/receive n/w packets. Update following values based on your system configuration.
+
+```
+diff --git a/scripts/docker_setup.sh b/scripts/docker_setup.sh
+index 7aff6a6..09d640b 100755
+--- a/scripts/docker_setup.sh
++++ b/scripts/docker_setup.sh
+@@ -24,7 +24,7 @@ mode="dpdk"
+ # Gateway interface(s)
+ #
+ # In the order of ("s1u" "sgi")
+-ifaces=("ens803f2" "ens803f3")
++ifaces=("enp134s0" "enp136s0")
+
+ # Static IP addresses of gateway interface(s) in cidr format
+ #
+@@ -34,7 +34,7 @@ ipaddrs=(198.18.0.1/30 198.19.0.1/30)
+ # MAC addresses of gateway interface(s)
+ #
+ # In the order of (s1u sgi)
+-macaddrs=(9e:b2:d3:34:ab:27 c2:9c:55:d4:8a:f6)
++macaddrs=(40:a6:b7:78:3f:ec 40:a6:b7:78:3f:e8)
+
+ # Static IP addresses of the neighbors of gateway interface(s)
+ #
+@@ -44,7 +44,7 @@ nhipaddrs=(198.18.0.2 198.19.0.2)
+ # Static MAC addresses of the neighbors of gateway interface(s)
+ #
+ # In the order of (n-s1u n-sgi)
+-nhmacaddrs=(22:53:7a:15:58:50 22:53:7a:15:58:50)
++nhmacaddrs=(40:a6:b7:78:3f:bc 40:a6:b7:78:3f:b8)
+```
+
+6) Modify the script [docker_setup.sh](../scripts/docker_setup.sh) and update the function `move_ifaces()` in condition `if [ "$mode" == 'cndp' ]` . Update `start_q_idx` to choose the start queue index to receive n/w packets. This should match the queue id used in lports section of [cndp_upf_1worker.jsonc](../conf/cndp_upf_1worker.jsonc). To get better performance (optional step), assign `cpuset-cpus` in [docker_setup.sh](../scripts/docker_setup.sh) to cores ids same as queue ids used to receive n/w packets.
```

Comment:
```suggestion
6) Modify the script [docker_setup.sh](../scripts/docker_setup.sh) and update the function `move_ifaces()` in condition `if [ "$mode" == 'cndp' ]`. Update `start_q_idx` to choose the start queue index to receive n/w packets. This should match the queue id used in lports section of [cndp_upf_1worker.jsonc](../conf/cndp_upf_1worker.jsonc). To get better performance (optional step), assign `cpuset-cpus` in [docker_setup.sh](../scripts/docker_setup.sh) to cores ids same as queue ids used to receive n/w packets.
```

Reply:
Will fix. How are you able to catch these issues without running any tool :-)

Reply:
No tool, just going through the entire PR :-)